services/api/mw2pdf/src/classes/MediaWikiSession.ts

```diff
@@ -1,131 +1,219 @@
+  isAuthenticated(domain: string): boolean {
+    return (domain in this.cookies);
+  }
 
   /**
    * Remove any stored authentication cookies.
    */
-  resetAuthentication() { this.cookies = {} }
+  resetAuthentication(): void {
+    this.cookies = new Map<string, Array<ParsedCookie>>();
+  }
 
-  getCookies(domain) { return this.cookies[domain] ?? [] }
+  getCookies(domain: string): Array<ParsedCookie> {
+    return this.cookies[domain] ?? [];
+  }
 
-  setCookies(domain, cookies) { this.cookies[domain] = cookies }
+  setCookies(domain: string, cookies: Array<ParsedCookie>): void {
+    this.cookies[domain] = cookies;
+  }
 
   /**
    * Authenticate against a given api URL.
    * If the authentication has already occurred for a given domain, this will NOT re-authenticate.
    */
-  async authenticate(username, password, apiUrl) {
-    const { host: domain } = new URL(apiUrl)
+  async authenticate(username: string, password: string, apiUrl: URL): Promise<boolean> {
+    const domain = apiUrl.host;
     if (this.isAuthenticated(domain)) {
       return true
     }
 
     // Step 1: Generate a login token
-    const loginTokenRequest = await makeLoginTokenRequest(apiUrl)
-    const authCookies = parseCookies(domain, loginTokenRequest)
-    const loginToken = await extractLoginToken(loginTokenRequest)
+    const loginTokenRequest = await makeLoginTokenRequest(apiUrl);
+    const authCookies = parseCookies(domain, loginTokenRequest);
+    const loginToken = await extractLoginToken(loginTokenRequest);
 
     // Step 2: Perform the login
-    const loginRequest = await makeLoginRequest({
+    const loginRequest = await makeLoginRequest(
       apiUrl,
       username,
       password,
       loginToken,
-      cookies: authCookies,
-    })
+      authCookies,
+    )
+
+    // Step 3: Set the cookies for the session
     this.setCookies(
       domain,
       parseCookies(domain, loginRequest),
     )
-    return true
+
+    return true;
   }
 
-  async makePdf(url, tmpDirectory = './', includeTitlePage = false) {
-    const pdfGenerator = new PdfGenerator(tmpDirectory)
+  async makePdf(url: URL, options: MwPdfOptions): Promise<Pdf> {
+    // Extract the domain (as _domain_) from the url.
+    const domain = url.host;
+
+    /**
+     * Begin puppeteer
+     */
+
+    // Open the browser
     const browser = await puppeteer.launch(
       { args: ['--no-sandbox', '--disable-setuid-sandbox'] }
-    )
-    const page = await browser.newPage()
-    const { host: domain } = new URL(url)
-    await page.setCookie(...this.getCookies(domain))
-    await page.goto(url, { waitUntil: 'networkidle2' })
+    );
+
+    // Get a new page
+    const page = await browser.newPage();
+
+    // Set cookies into the browser that we pull from the MW session. This means
+    // that all the login and authentication we did before will be valid.
+    await page.setCookie(...this.getCookies(domain));
+
+    // Now, go to the page!
```

Comment:
There are some other comments that are redundant (which is a good problem to have -- it means we named our variables / methods very well!

Reply:
I removed the ones that I thought were redundant. Let me know what you think!